easybuild/easyblocks/generic/juliapackage.py

```diff
@@ -0,0 +1,170 @@
+"""
+import os
+
+from distutils.version import LooseVersion
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.framework.extensioneasyblock import ExtensionEasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.filetools import copy_dir
+from easybuild.tools.run import run_cmd
+
+EXTS_FILTER_JULIA_PACKAGES = ("julia -e 'using %(ext_name)s'", "")
+
+
+class JuliaPackage(ExtensionEasyBlock):
+    """Builds and installs Julia Packages."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to JuliaPackage."""
+        extra_vars = ExtensionEasyBlock.extra_options(extra_vars=extra_vars)
+        extra_vars.update({
+            'bundle_deps': [
+                False, "Let Julia download and bundle all needed dependencies for this installation", CUSTOM
+            ],
+        })
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """JuliaPackage easyblock constructor."""
+        super(JuliaPackage, self).__init__(*args, **kwargs)
+
+    def set_pkg_offline(self):
+        """Enable offline mode of Julia Pkg"""
+        if get_software_root('Julia') is None:
+            raise EasyBuildError("Julia not included as dependency!")
+
+        if not self.cfg['bundle_deps']:
+            julia_version = get_software_version('Julia')
+            if LooseVersion(julia_version) >= LooseVersion('1.5'):
+                # Enable offline mode of Julia Pkg
+                # https://pkgdocs.julialang.org/v1/api/#Pkg.offline
+                env.setvar('JULIA_PKG_OFFLINE', 'true')
+            else:
+                errmsg = (
+                    "Cannot set offline mode in Julia v%s (needs Julia >= 1.5). "
+                    "Enable easyconfig option 'bundle_deps' to allow installation "
+                    "with any extra downloaded dependencies."
+                )
+                raise EasyBuildError(errmsg, julia_version)
+
+    def prepare_step(self, *args, **kwargs):
+        """Prepare for installing Julia package."""
+        super(JuliaPackage, self).prepare_step(*args, **kwargs)
+        self.set_pkg_offline()
+
+    def configure_step(self):
+        """No separate configuration for JuliaPackage."""
+        pass
+
+    def build_step(self):
+        """No separate build procedure for JuliaPackage."""
+        pass
+
+    def test_step(self):
+        """No separate (standard) test procedure for JuliaPackage."""
+        pass
+
+    def install_julia_pkg(self):
+        """Install procedure for Julia packages with Pkg"""
+
+        # prepend installation directory to Julia DEPOT_PATH
+        # see https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_DEPOT_PATH
+        env.setvar('JULIA_DEPOT_PATH', ':'.join([self.installdir, os.getenv('JULIA_DEPOT_PATH', '')]))
```

Comment:
Maybe you want to avoid a trailing `:` here:

```python
depot_path = os.getenv('JULIA_DEPOT_PATH')
if depot_path is not None:
    depot_path = ':'.join([self.installdir, depot_path])
env.setvar('JULIA_DEPOT_PATH', depot_path)

Reply:
good point, fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2816/commits/744a628a8928de76ffbf4bb2e29d5aadd6c2c5e9